### PR TITLE
Create custom Xcode schemes

### PR DESCRIPTION
**Pull request by cgrindel**
Related to #573.

- Add initializers with defaults to `XcodeScheme` structs.
- Add `XcodeScheme.ProfileAction`.
- Move `shouldCreateScheme` filter to `createAutogeneratedXCSchemes`. It was in the shared code.
- Add `targetResolver(referencedContainer:)` to `Fixtures`.

### CODE REVIEW

File: tools/generator/src/Extensions/Dictionary+Extensions.swift
Comment (brentleyjones) [inline tools/generator/src/Extensions/Dictionary+Extensions.swift:2]: Can you add this in its own PR, along with changing code to use this new method?

File: tools/generator/src/Extensions/XCScheme+Extensions.swift
Comment (brentleyjones) [inline tools/generator/src/Extensions/XCScheme+Extensions.swift:12]: I prefer refactor and behavior changes to be in separate PRs. Can you make this change in its own PR?

Comment (cgrindel) [reply tools/generator/src/Extensions/XCScheme+Extensions.swift:12]: #804

File: tools/generator/src/Extensions/Dictionary+Extensions.swift
Comment (cgrindel) [reply tools/generator/src/Extensions/Dictionary+Extensions.swift:2]: I have added the new method in #805. I am happy to help with switching code over to use this new method. However, I don't think that this should hold up this PR.

**Review by brentleyjones — DISMISSED** [review]
If we are creating custom schemes, shouldn't the ones for Generator have changed?

**Comment by cgrindel** [comment]
@brentleyjones You are correct. I will investigate before merging this PR.

**Comment by cgrindel** [comment]
The `//tools/generator:xcodeproj` target is configured with `scheme_autogeneration_mode` set to `all`. Hence, it is generating both the custom and autogenerated scheme. The custom scheme is named `Generator`. One of the autogenerated schemes is called `generator`. The last one in the list wins.

For this PR, I will rename the custom scheme to not conflict. I have added two tasks to #573. 
- Enable `scheme_autogenrate_mode` `auto`.
- Ensure that autogenerated scheme names do not conflict with custom scheme names.

**Comment by cgrindel** [comment]
I verified that the custom scheme is written (Good catch, @brentleyjones). I launched the scheme in Xcode and was able to build and test it successfully.

**Review by brentleyjones — APPROVED** [review]
When there is conflicting schemes, can we have the custom ones just override the automatic one? Seems better than an error, since one controls the name of the custom scheme.

**Comment by brentleyjones** [comment]
(Would require changing the order that they are made in, but I think that's for the best.)

**Comment by cgrindel** [comment]
> When there is conflicting schemes, can we have the custom ones just override the automatic one? Seems better than an error, since one controls the name of the custom scheme.

Sure. I was going to add code that checks an autogenerated scheme name and add a prefix (e.g., `(generated)`), if there was a conflict. However, I am happy to just skip the autogenerated one. Whichever you want is fine with me.

**Comment by brentleyjones** [comment]
I like skipping the autogenerated one. Then one can use `top_level` and still define a custom one to "tweak" and existing top level autogenerated scheme.

**Pull request merged**
